extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/linkHandler/YoutubeStreamLinkHandlerFactory.java

```diff
@@ -15,41 +35,21 @@
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URL;
-import java.util.Arrays;
-import java.util.List;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-/*
- * Created by Christian Schabesberger on 02.02.16.
- *
- * Copyright (C) Christian Schabesberger 2018 <chris.schabesberger@mailbox.org>
- * YoutubeStreamLinkHandlerFactory.java is part of NewPipe.
- *
- * NewPipe is free software: you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation, either version 3 of the License, or
- * (at your option) any later version.
- *
- * NewPipe is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with NewPipe.  If not, see <http://www.gnu.org/licenses/>.
- */
-
 public final class YoutubeStreamLinkHandlerFactory extends LinkHandlerFactory {
 
     private static final Pattern YOUTUBE_VIDEO_ID_REGEX_PATTERN
             = Pattern.compile("^([a-zA-Z0-9_-]{11})");
     private static final YoutubeStreamLinkHandlerFactory INSTANCE
             = new YoutubeStreamLinkHandlerFactory();
-    private static final List<String> SUBPATHS
-            = Arrays.asList("embed/", "shorts/", "watch/", "v/", "w/");
+    private static final Set<String> SUBPATHS
+            = Set.of("embed/", "live/", "shorts/", "watch/", "v/", "w/");
```

Comment:
You can also create unmodifiable lists using the same method in `List`. What the reasoning for switching to a set?

Reply:
I thought that using sets would be better, but it doesn't seem to be the case for our usage. I changed `SUBPATHS` type back to `List`.